session/memory.js

```diff
@@ -32,12 +33,47 @@ module.exports = MemoryStore
 
 /**
  * A session store in memory.
+
+ * @param {Object} [options]
+ * @param {Number} [options.timeout] Session timeout (in minutes). Use a negative value to never expire.
  * @public
  */
 
-function MemoryStore() {
+function MemoryStore(options) {
   Store.call(this)
   this.sessions = Object.create(null)
+  this.session_timers = Object.create(null)
+
+  // set up the store
+  this.options = {
+    timeout: -1 // sessions never expire by default
+  }
+
+  if (options) {
+    if (typeof options.timeout !== 'undefined') {
+      this.options.timeout = options.timeout
+    }
+  }
+
+  debug('session timeout is set to ' + this.options.timeout)
+
+  // run the session timeout checker
+  if (this.options.timeout >= 0) {
+    var self = this
+    this.session_timeout_checker = setInterval(function() {
+      debug('checking sessions for timeout')
+      for (var sessionId in self.session_timers) {
+        if (self.session_timers[sessionId] === 0) {
+          debug('session ' + sessionId + ' timed out, destroying it')
+          self.destroy(sessionId)
+        }
+        else {
+          self.session_timers[sessionId]--
```

Comment:
@saperal indeed, `setInterval` doesn't fire immediately, therefore it should be

```javascript
+      for (var sessionId in self.session_timers) {
+        self.session_timers[sessionId]--
+        if (self.session_timers[sessionId] <= 0) {
+          debug('session ' + sessionId + ' timed out, destroying it')
+          self.destroy(sessionId)
+        }
```

Seems that noone is interested in merging this PR tho (it has already been 3 months).